test_tools/example_gen/example_gen.cc

```diff
@@ -0,0 +1,181 @@
+  config.set("observation.sender.implementation", "OBSERVATION_FILE_SENDER");
+  config.set("decisions.sender.implementation", "INTERACTION_FILE_SENDER");
+  config.set("model.source", "NO_MODEL_DATA");
+
+  bool is_observation = action >= F_REWARD;
+  if(is_observation) {
+    config.set("observation.file.name", file_name.c_str());
+    config.set("interaction.file.name", "/dev/null");
+  } else {
+    config.set("observation.file.name", "/dev/null");
+    config.set("interaction.file.name", file_name.c_str());
+  }
+  config.set("protocol.version", "2");
+  config.set("InitialExplorationEpsilon", "1.0");
+
+  if(action == CCB_ACTION) {
+    config.set(r::name::MODEL_VW_INITIAL_COMMAND_LINE, "--ccb_explore_adf --json --quiet --epsilon 0.0 --first_only --id N/A");
+  } else if (action == SLATES_ACTION) {
+    config.set(r::name::MODEL_VW_INITIAL_COMMAND_LINE, "--slates --ccb_explore_adf --json --quiet --epsilon 0.0 --first_only --id N/A");
+  }
+}
+
+const auto JSON_CB_CONTEXT = R"({ "GUser":{"id":"a","major":"eng","hobby":"hiking"}, "_multi":[ { "TAction":{"a1":"f1"} },{"TAction":{"a2":"f2"}}]})";
+
+const auto JSON_CCB_CONTEXT = R"({"GUser":{"id":"a","major":"eng","hobby":"hiking"},"_multi":[ { "TAction":{"a1":"f1"} },{"TAction":{"a2":"f2"}}],"_slots":[{"Slot":{"a1":"f1"}},{"Slot":{"a1":"f1"}}]})";
+
+const auto JSON_SLATES_CONTEXT = R"({"GUser":{"id":"a","major":"eng","hobby":"hiking"},"_multi":[{"TAction":{"a1":"f1"},"_slot_id":0},{"TAction":{"a2":"f2"},"_slot_id":0},{"TAction":{"a3":"f3"},"_slot_id":1},{"TAction":{"a4":"f4"},"_slot_id":1},{"TAction":{"a5":"f5"},"_slot_id":1}],"_slots":[{"Slot":{"a1":"f1"}},{"Slot":{"a2":"f2"}}]})";
+
+
+int run_config(int action) {
+  u::configuration config;
+
+  load_config_from_json(action, config);
+
+  r::api_status status;
+  r::live_model rl(config);
+  if( rl.init(&status) != err::success ) {
+    std::cout << status.get_error_msg() << std::endl;
+    return -1;
+  }
+
+  switch(action) {
+    case CB_ACTION: {// "cb",
+      r::ranking_response response;
+      rl.choose_rank("event_id", JSON_CB_CONTEXT, response, &status);
+      break;
+    }
+    case CCB_ACTION: {// "ccb",
+      r::slates_response response;
+      if(rl.request_slates_decision("event_id", JSON_CCB_CONTEXT, 0, response, &status) != err::success)
+          std::cout << status.get_error_msg() << std::endl;
+      break;
+    };
+    case SLATES_ACTION: {// "slates",
+      r::slates_response response;
+      if(rl.request_slates_decision("event_id", JSON_SLATES_CONTEXT, response, &status) != err::success)
+          std::cout << status.get_error_msg() << std::endl;
+      break;
+    };
+    case F_REWARD: // "float"
+      if( rl.report_outcome("event_id", 1.5, &status) != err::success ) 
+          std::cout << status.get_error_msg() << std::endl;
+      break;
+
+    case F_I_REWARD: // "float-int",
+      if( rl.report_outcome("event_id", 1.5, 10, &status) != err::success ) 
```

Comment:
Is this supposed to go to the `report_outcome(const char* primary_id, int secondary_id, float outcome, api_status* status)` method? Are the arguments supposed to be reversed here, i.e. `"event_id", 10, 1.5`?